c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -624,8 +624,21 @@ MfxC2DecoderComponent::MfxC2DecoderComponent(const C2String name, const CreateCo
         .withSetter(ColorAspectsSetter, m_defaultColorAspects, m_codedColorAspects)
         .build());
 
-    // Pixel format info. Set to NV12 by default
-    m_pixelFormat = std::make_unique<C2StreamPixelFormatInfo::output>(SINGLE_STREAM_ID, HAL_PIXEL_FORMAT_NV12_Y_TILED_INTEL);
+    std::vector<uint32_t> supportedPixelFormats = {
+        HAL_PIXEL_FORMAT_YCBCR_420_888,
+        HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED,
+        HAL_PIXEL_FORMAT_YCRCB_420_SP,
```

Comment:
Can you explain why HAL_PIXEL_FORMAT_NV12_Y_TILED_INTEL is removed here?

Reply:
Can we declare 10bit format as HAL_PIXEL_FORMAT_P010_INTEL here?

Thanks,
Tianmi

Reply:
I think framework should not recognize the formats with "INTEL" in the name.
http://inlubt0232.iind.intel.com:8080/source/xref/CELADON/frameworks/av/media/codec2/sfplugin/utils/Codec2Mapper.cpp?r=e9707b37#440

Reply:
Yes, framework use HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED to cover all vendor supported formats. Although framework doesn't care about it, we know the detailed info of our supported yuv format in codec2.0 HAL. I think we should expose intel specific color formats HAL_PIXEL_FORMAT_NV12_Y_TILED_INTEL and HAL_PIXEL_FORMAT_P010_INTEL for decoder, and some RGB( like HAL_PIXEL_FORMAT_RGBA_8888) and YUV formats for encoder.

Reply:
I tried to use HAL_PIXEL_FORMAT_P010_INTEL instead of HAL_PIXEL_FORMAT_YCBCR_P010 on this case, but failed. If we want, we can add those types in supportedFormats, but I think it may be useless.

Reply:
what's the error if we use HAL_PIXEL_FORMAT_P010_INTEL for this av1 case?

Reply:
because they don't recognize HAL_PIXEL_FORMAT_P010_INTEL, if we set it to supported format, they think we don't support 10-bit format, they will use the default 8-bit formt to configure codec param, but the source video is a 10-bit sample.